qt/scientific_interfaces/test/ISISReflectometry/MainWindow/MainWindowPresenterTest.h

```diff
@@ -396,13 +413,15 @@ class MainWindowPresenterTest : public CxxTest::TestSuite {
     auto presenter = MainWindowPresenterFriend(
         &m_view, &m_messageHandler, &m_fileHandler, std::move(encoder),
         std::move(decoder), std::move(slitCalculator),
-        std::move(makeBatchPresenter));
+        std::move(optionsDialogPresenter), std::move(makeBatchPresenter));
     return presenter;
   }
 
-  void verifyAndClear() {
+  void verifyAndClear(
+      NiceMock<MockOptionsDialogPresenter> *optionsDialogPresenter = nullptr) {
     TS_ASSERT(Mock::VerifyAndClearExpectations(&m_view));
     TS_ASSERT(Mock::VerifyAndClearExpectations(&m_messageHandler));
+    TS_ASSERT(Mock::VerifyAndClearExpectations(&optionsDialogPresenter));
```

Comment:
This is already a pointer so you don't want to take the address - it actually doesn't do the verification of the correct object at the moment so the tests are not verifying it!
```suggestion
    TS_ASSERT(Mock::VerifyAndClearExpectations(optionsDialogPresenter));
```